### PR TITLE
tolower/upper conversions

### PR DESCRIPTION
**Pull request by neheb**

### CODE REVIEW

**Comment by gizmocuz** [comment]
Thank you for this PR. Could you undo the changes in webserver/request_parser.cpp please

**Comment by neheb** [comment]
Done.

**Comment by neheb** [comment]
Rebased. From 10069656 to 9918104 locally.

File: hardware/1Wire/1WireByKernel.cpp [hunk 1]
```diff
@@ -551,7 +543,8 @@ void C1WireByKernel::GetDevice(const std::string& deviceName, /*out*/_t1WireDevi
 	device.family = ToFamily(deviceName.substr(0, 2));
 
 	// Device Id (6 chars after '.')
-	std_to_upper(deviceName.substr(3, 3 + 6 * 2), device.devid);
+	const auto str = deviceName.substr(3, 3 + 6 * 2);
+	std::transform(str.begin(), str.end(), device.devid.begin(), ::toupper);
```
Comment (gizmocuz) [inline hardware/1Wire/1WireByKernel.cpp]: We do have a 'stdupper' upper function in the helper.cpp/h file

Comment (neheb) [reply hardware/1Wire/1WireByKernel.cpp]: That's not going to work. Source and destination are not the same.

Comment (neheb) [reply hardware/1Wire/1WireByKernel.cpp]: Eh kind of did it. Requires one more line :).

File: hardware/1Wire/1WireByKernel.cpp [hunk 2]
```diff
@@ -551,7 +543,9 @@ void C1WireByKernel::GetDevice(const std::string& deviceName, /*out*/_t1WireDevi
 	device.family = ToFamily(deviceName.substr(0, 2));
 
 	// Device Id (6 chars after '.')
-	std_to_upper(deviceName.substr(3, 3 + 6 * 2), device.devid);
+	auto str = deviceName.substr(3, 3 + 6 * 2);
```
Comment (gizmocuz) [inline hardware/1Wire/1WireByKernel.cpp:546]: Could this be declared as std::string

Comment (neheb) [reply hardware/1Wire/1WireByKernel.cpp:546]: auto deduces to std::string.

Comment (gizmocuz) [reply hardware/1Wire/1WireByKernel.cpp:546]: Yep, and this way we can use auto everywhere.
I vote readability over optimization

Comment (gizmocuz) [reply hardware/1Wire/1WireByKernel.cpp:546]: Could you revert this change and declare it as std::string ?

**Pull request closed without merging**